ClientRuntimes/NodeJS/ClientRuntime/test/redirectfilterTests.js

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+var assert = require('assert');
+var url = require('url');
+var should = require('should');
+
+var RedirectFilter = require('../lib/filters/redirectfilter');
+
+describe('Redirect filter', function () {
+  it('should redirect a POST request with status code 303 into a GET request when redirect url is present in location header', function (done) {
+    var resource = { method: 'POST', url: 'http://localhost:3000/http/redirect/303' };
+    var response1 = { statusCode: 303, body: '', headers: { location: '/http/success/get/200' }, method: 'POST' };
+    var redirectFilter = RedirectFilter.create();
+    
+    var mocknext = function (resource, handleRedirect) {
+      if (resource.method !== 'GET') {
+        handleRedirect(null, response1, response1.body);
+      } else {
+        resource.method.should.equal('GET');
+        resource.url.should.equal('http://localhost:3000/http/success/get/200');
+        done();
+      }
+    };
+    
+    redirectFilter(resource, mocknext, null);
+  });
+  
+  it('should not redirect a POST request with status code 303 into a GET request when redirect url is not present in location header', function (done) {
+    var resource = { method: 'POST', url: 'http://localhost:3000/http/redirect/303' };
+    var response1 = { statusCode: 303, body: '', headers: {}, method: 'POST' };
```

Comment:
Based on the coding guideline, though test code, we should still apply line break for more readable.
https://github.com/Azure/azure-coding-guidelines/blob/master/nodejs.md